File: araok/src/main/java/ru/otus/dto/ContentDto.java
package ru.otus.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.otus.domain.Content;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ContentDto {
    private Long id;

    private String name;

    private AgeLimitDto limit;

    private String artist;

    private UserDto user;

    private LocalDate createDate;

    private LanguageDto language;

    public static Content toDomainObject(ContentDto dto) {
        return Content.builder()
                .id(dto.id)
                .name(dto.name)
                .limit(
                        AgeLimitDto.toDomainObject(dto.limit)
                )
                .artist(dto.artist)
                .user(
                        UserDto.toDomainObject(dto.user)
                )
                .createDate(dto.createDate)
                .language(
                        LanguageDto.toDomainObject(dto.language)
                )
                .build();
    }

    public static ContentDto toDto(Content content) {
        return ContentDto.builder()
                .id(content.getId())
                .name(content.getName())
                .limit(
                        AgeLimitDto.toDto(content.getLimit())
                )
                .artist(content.getArtist())
                .user(
                        UserDto.toDto(content.getUser())
                )
                .createDate(content.getCreateDate())
                .language(
                        LanguageDto.toDto(content.getLanguage())
                )
                .build();
    }
}

File: araok-android/src/main/java/ru/otus/data/dto/ContentDto.kt
package ru.araok.data.dto

import android.os.Build
import androidx.annotation.RequiresApi
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import ru.araok.entites.Content
import java.time.LocalDate

@JsonClass(generateAdapter = true)
data class ContentDto @RequiresApi(Build.VERSION_CODES.O) constructor(
    @Json(name = "id") override val id: Long? = null,
    @Json(name = "name") override val name: String = "",
    @Json(name = "limit") override val limit: AgeLimitDto = AgeLimitDto(id = null, description = "", limit = 0),
    @Json(name = "artist") override val artist: String = "",
    @Json(name = "user") override val user: UserDto = UserDto(id = 0, name = "", phone = "", password = "", LocalDate.now(), role = ""),
    @Json(name = "createDate") override val createDate: LocalDate = LocalDate.now(),
    @Json(name = "language") override val language: LanguageDto = LanguageDto(id = null, language = "", code2 = "")
): Content
